obspy/io/mseed/util.py

```diff
@@ -238,31 +275,110 @@ def get_timing_and_data_quality(file_or_file_object):
+        elif starttime is not None:
+            if this_info['starttime'] >= starttime:
+                if(io_flags):
+                    # Add the value of each bit to the io_count.
+                    for _i in range(6):
+                        if (this_info['io_and_clock_flags'] & (1 << _i)) != 0:
+                            io_count[_i] += 1
+                if(activity_flags):
+                    # Add the value of each bit to the activity_count.
+                    for _i in range(7):
+                        if (this_info['activity_flags'] & (1 << _i)) != 0:
+                            activity_count[_i] += 1
+                # Add the timing quality.
+                if 'timing_quality' in this_info:
+                    timing_quality.append(float(this_info['timing_quality']))
+                # Add the value of each bit to the quality_count.
+                for _i in range(8):
+                        if (this_info['data_quality_flags'] & (1 << _i)) != 0:
+                            quality_count[_i] += 1
+        elif endtime is not None:
+            if this_info['starttime'] <= endtime:
+                if(io_flags):
+                    # Add the value of each bit to the io_count.
+                    for _i in range(6):
+                        if (this_info['io_and_clock_flags'] & (1 << _i)) != 0:
+                            io_count[_i] += 1
+                if(activity_flags):
+                    # Add the value of each bit to the activity_count.
+                    for _i in range(7):
+                        if (this_info['activity_flags'] & (1 << _i)) != 0:
+                            activity_count[_i] += 1
+                # Add the timing quality.
+                if 'timing_quality' in this_info:
+                    timing_quality.append(float(this_info['timing_quality']))
+                # Add the value of each bit to the quality_count.
+                for _i in range(8):
+                        if (this_info['data_quality_flags'] & (1 << _i)) != 0:
+                            quality_count[_i] += 1
+        else:
+            if(io_flags):
+                # Add the value of each bit to the io_count.
+                for _i in range(6):
+                    if (this_info['io_and_clock_flags'] & (1 << _i)) != 0:
+                        io_count[_i] += 1
+            if(activity_flags):
+                # Add the value of each bit to the activity_count.
+                for _i in range(7):
+                    if (this_info['activity_flags'] & (1 << _i)) != 0:
+                        activity_count[_i] += 1
+            # Add the timing quality.
+            if 'timing_quality' in this_info:
+                timing_quality.append(float(this_info['timing_quality']))
+            # Add the value of each bit to the quality_count.
+            for _i in range(8):
+                    if (this_info['data_quality_flags'] & (1 << _i)) != 0:
+                        quality_count[_i] += 1
```

Comment:
I agree. That is still be the case with my modified version.

The other question to ask is what happens when you set `starttime` to like 5, and a record goes from 0 to 10. Do we still count that record or only the next one? Same with the end time setting.